src/callbacks_step/save_solution.jl

```diff
@@ -6,23 +6,26 @@
 
 
 """
-    SaveSolutionCallback(; interval=0,
+    SaveSolutionCallback(; interval::Integer=0, dt=0.0,
                            save_initial_solution=true,
                            save_final_solution=true,
                            output_directory="out",
                            solution_variables=cons2prim)
 
-Save the current numerical solution every `interval` time steps. `solution_variables` can be any
-callable that converts the conservative variables at a single point to a set of solution variables.
-The first parameter passed to `solution_variables` will be the set of conservative variables and the
-second parameter is the equation struct.
+Save the current numerical solution in regular intervals. Either pass `interval` to save
+every `interval` time steps, or pass `dt` to save in intervals of `dt` in terms
+of integration time by adding additional `tstops` (note that this may change the solution).
+`solution_variables` can be any callable that converts the conservative variables
+at a single point to a set of solution variables. The first parameter passed
+to `solution_variables` will be the set of conservative variables
+and the second parameter is the equation struct.
 """
-mutable struct SaveSolutionCallback{SolutionVariables}
-  interval::Int
+mutable struct SaveSolutionCallback{I, SV}
+  interval::I
   save_initial_solution::Bool
   save_final_solution::Bool
   output_directory::String
-  solution_variables::SolutionVariables
+  solution_variables::SV
```

Comment:
Why?

Reply:
I was confused where `SolutionVariables` was defined because it looked like a type name.

Reply:
Please see the comment of Michael above. I also vote for a more descriptive name.